### PR TITLE
Create new specialist document type: Product safety alert

### PR DESCRIPTION
**Pull request by ChrisBAshton**
This is for the new "Product safety alerts, unsafe products and
recalls" finder. I've followed https://docs.publishing.service.gov.uk/apps/specialist-publisher/creating-a-new-specialist-document-type.html,
but also taken inspiration from the most recent PR #1910.

Screenshots:

## Index page

> Screenshot 2021-10-29 at 10 50 27

## New page

> Screenshot 2021-10-29 at 10 52 15

Trello: https://trello.com/c/ojsFViiX/2746-create-new-specialist-document-and-finder-product-safety-alerts-unsafe-products-and-recalls-3

### CODE REVIEW

File: app/models/product_safety_alert.rb
Comment (1pretz1) [inline app/models/product_safety_alert.rb]: Not sure we need to define the `taxons` method here as the parent model has the same method, with the same output (empty array): https://github.com/alphagov/specialist-publisher/blob/main/app/models/document.rb#L84-L86.

File: lib/documents/schemas/product_safety_alerts.json
Comment (1pretz1) [inline lib/documents/schemas/product_safety_alerts.json:15]: I've just looked at the associated card and it mentions that there should (?) be a subscription option for multiple facets, rather than just the 'Alert type', as per:

```
**Facets for the Finder:**
Users need to be able to filter and/or subscribe by :
- Alert Type
- Risk Level
- Product Category
- Measure Type
- Recall/alert date (before and after picker, like https://www.gov.uk/drug-device-alerts)
```

The comment isn't really clear on what subscription options are needed (and/or). Has there been an active choice here?

**Review by 1pretz1 — COMMENTED** [review]
Looks good @ChrisBAshton - thanks for providing screenshots 📸 

Just have one minor suggestion and a query.

File: lib/documents/schemas/product_safety_alerts.json
Comment (ChrisBAshton) [reply lib/documents/schemas/product_safety_alerts.json:15]: Great spot - thanks @1pretz1! I've pushed up a commit that I think fixes that (lack of documentation, but I've explained my reasoning in the commit). Would be great to get a second pair of eyes on that.
 
I _think_ the new commit means we can close https://github.com/alphagov/email-alert-api/pull/1672 (based on https://docs.publishing.service.gov.uk/apps/specialist-publisher/creating-a-new-specialist-document-type.html#configure-the-email-sign-up-page) 🤔

Comment (1pretz1) [reply lib/documents/schemas/product_safety_alerts.json:15]: Ah, nice! The `all_selected_facets` option does seem less user friendly though as the user doesn't get a check box / confirmation of what facets they're subscribing to (until they've actually subscribed), for example https://www.gov.uk/residential-property-tribunal-decisions/email-signup (all facets option) vs https://www.gov.uk/cma-cases/email-signup. I think this happens due to the [logic in Finder Frontend](https://github.com/alphagov/finder-frontend/blob/main/app/presenters/subscriber_list_params_presenter.rb#L36-L38).

However, if want to enable email subscriptions for all the facets then this seems the only option. Hmm, should we ask BEIS to clarify if they actually want all facets, or just go with the `all_selected_facets`?

> I think the new commit means we can close alphagov/email-alert-api#1672

I believe you're correct. Could we also remove the `email_filter_facets` hash from this schema if we use `all_selected_facets`?

Comment (ChrisBAshton) [reply lib/documents/schemas/product_safety_alerts.json:15]: Good shout. We're checking with BEIS. On hold until then.

Comment (ChrisBAshton) [reply lib/documents/schemas/product_safety_alerts.json:15]: BEIS have decided to go with the standard `product_alert_type`. Could you re-review please, @1pretz1 ?

**Review by 1pretz1 — APPROVED** [review]
LGTM! 🍏

**Pull request merged**
